### PR TITLE
simplify and speedup NaNCheck

### PR DESCRIPTION
**Pull request by ChrisRackauckas**
From https://discourse.julialang.org/t/fastest-way-to-check-for-inf-or-nan-in-an-array/76954/22 @oscardssmith

### CODE REVIEW

File: src/common_defaults.jl [hunk 1]
```diff
@@ -10,38 +10,35 @@ abs2_and_sum(x,y) = reduce(Base.add_sum,x,init=zero(real(value(eltype(x))))) +
...
-@inline NAN_CHECK(x::Number) = isnan(x)
-@inline NAN_CHECK(x::Float64) = isnan(x) || (x>1e50)
-@inline NAN_CHECK(x::Enum) = false
-@inline NAN_CHECK(x::AbstractArray) = any(NAN_CHECK, x)
-@inline NAN_CHECK(x::RecursiveArrayTools.ArrayPartition) = any(NAN_CHECK, x.x)
 
-@inline ODE_DEFAULT_UNSTABLE_CHECK(dt,u,p,t) = false
-@inline ODE_DEFAULT_UNSTABLE_CHECK(dt,u::Union{Number,AbstractArray},p,t) = NAN_CHECK(u)
+@inline NAN_CHECK(x::Number) = !isfinite(x)
+@inline NAN_CHECK(x::AbstractArray) = !isfinite(sum(x))
```
Comment (oscardssmith) [inline src/common_defaults.jl:42]: isn't this wrong? (sum can overflow)

Comment (ChrisRackauckas) [reply src/common_defaults.jl:42]: Truth

**Review by YingboMa — APPROVED** [review]
LGTM. But maybe someone’s solution is actually that big?

**Comment by YingboMa** [comment]
Is it actually faster?

**Pull request closed without merging**